config/initializers/health_checks.rb

```diff
@@ -0,0 +1,24 @@
+# okcomputer is used for health checks
+
+username = ENV["HEALTHCHECK_HTTP_BASIC_AUTH_USERNAME"] || ENV["HTTP_BASIC_AUTH_USERNAME"]
+password = ENV["HEALTHCHECK_HTTP_BASIC_AUTH_PASSWORD"] || ENV["HTTP_BASIC_AUTH_PASSWORD"]
+OkComputer.require_authentication(username, password) if username && password
+
+# Check that Redis is available
+OkComputer::Registry.register "app_version", OkComputer::AppVersionCheck.new
+
+# Additional checks can be added, e.g.
+# OkComputer::Registry.register "mailing", OkComputer::ActionMailerCheck.new
+# OkComputer::Registry.register "redis", OkComputer::RedisCheck.new({})
+# OkComputer::Registry.register "sidekiq_default", OkComputer::SidekiqLatencyCheck.new(:default)
+# OkComputer::Registry.register "sidekiq_mailers", OkComputer::SidekiqLatencyCheck.new(:mailers)
```

Comment:
These sidekiq ones make me a little squeamish. On the one hand it could be useful to know but on the other hand, load balancers will hit this check and make decisions based on it and I'm not sure that in all situations we would want Sidekiq latency to trigger cycling an EC2 instance? 

I'm not sure whether this squeamishness is enough to make me want to remove those comments or whether we should put a warning comment above them that a dev shouldn't just blithely enable them if they have sidekiq. Thoughts?

Reply:
@eoinkelly I think the right thing to do here would be to declare the Sidekiq, Mailers, and maybe the Redis checks as optional, using: `OkComputer.make_optional %w(some_optional_check another_optional_check)`

This would mean that the healthchecks would still report, but would not cause the overall check to fail.

Reply:
:thumbsup: I think if we update the comments to use `make_optional` instead of `register` this this is sorted.

Reply:
Ah I see that `register` has to happen anyway and they can still be made optional